src/main/java/com/aws/greengrass/shadowmanager/sync/IotDataPlaneClientFactory.java

```diff
@@ -63,12 +73,13 @@ public class IotDataPlaneClientFactory {
      */
     @Inject
     public IotDataPlaneClientFactory(DeviceConfiguration deviceConfiguration) {
-        configureClient(deviceConfiguration);
+        this.deviceConfiguration = deviceConfiguration;
+        configureClient(true);
```

Comment:
this may cause blocking in the constructor! We must not block

Reply:
Removed.